### PR TITLE
Renamed and restructured OpenGL files and folders.

### PR DESCRIPTION
**Pull request by jonathanudd**

### CODE REVIEW

**Comment by jonathanudd** [comment]
@emilwestergren can you check if this seems ok?

**Comment by emilwestergren** [comment]
Looks good but I'm not sure about the OpenGLImplementation folder. It should probably be named *backend* or something, you should ask @simonmika about this.

**Comment by davidhesselbom** [comment]
Conflicts...

File: source/draw/gpu/opengl/backend/gles3/DebugGL.ooc [hunk 1]
```diff
@@ -1,5 +1,6 @@
 use ooc-base
-import include/[egl, gles]
+import ../egl/egl
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/backend/gles3/DebugGL.ooc]: This looks weird. @simonmika, can importing from `../whatever` be avoided?

File: source/draw/gpu/opengl/OpenGLBgra.ooc [hunk 2]
```diff
@@ -18,11 +18,11 @@
 use ooc-math
 use ooc-draw
 use ooc-draw-gpu
-import OpenGLES3/Texture, OpenGLES3Canvas, OpenGLES3Texture
+import backend/gles3/Texture, OpenGLCanvas, OpenGLTexture
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/OpenGLBgra.ooc]: So the plan is to remove this import as well soon, or what?

File: test/draw/gpu/pc/GpuSurfaceTest.ooc [hunk 3]
```diff
@@ -139,7 +139,7 @@ GpuSurfaceTest: class extends Fixture {
 			gpuImage canvas drawLines(trianglePoints)
 			gpuImage canvas drawBox(FloatBox2D new(-lineLength, -lineLength, lineLength * 2.0f, lineLength * 2.0f))
 			//
-			// NOTE! The circle use a point size of 1.0f (OpenGLES3MapPoints in OpenGLES3Map.ooc)
+			// NOTE! The circle use a point size of 1.0f (OpenGLMapPoints in OpenGLMap.ooc)
```
Comment (davidhesselbom) [inline test/draw/gpu/pc/GpuSurfaceTest.ooc]: Just checking - Is this comment still valid?

File: source/draw/gpu/android/AndroidContext.ooc [hunk 4]
```diff
@@ -22,11 +22,10 @@ use ooc-base
 import AndroidTexture, GraphicBuffer
 import threading/Thread
 import math
-
-AndroidContext: class extends OpenGLES3Context {
-	_unpackRgbaToMonochrome := OpenGLES3MapUnpackRgbaToMonochrome new(this)
-	_unpackRgbaToUv := OpenGLES3MapUnpackRgbaToUv new(this)
-	_packers := VectorList<OpenGLES3Bgra> new()
+AndroidContext: class extends OpenGLContext {
```
Comment (davidhesselbom) [inline source/draw/gpu/android/AndroidContext.ooc]: Don't remove the empty line here, please.

File: source/draw/gpu/android/AndroidContext.ooc [hunk 5]
```diff
@@ -22,11 +22,10 @@ use ooc-base
 import AndroidTexture, GraphicBuffer
 import threading/Thread
 import math
```
Comment (davidhesselbom) [inline source/draw/gpu/android/AndroidContext.ooc]: Don't remove the empty line here, please.

File: source/draw/gpu/opengl/backend/gles3/Context.ooc [hunk 6]
```diff
@@ -16,8 +16,7 @@
  */
 use ooc-base
 use ooc-math
-import include/[egl, gles]
-import NativeWindow, DebugGL
+import ../egl/egl, include/gles3, NativeWindow, DebugGL
```
Comment (davidhesselbom) [inline source/draw/gpu/opengl/backend/gles3/Context.ooc]: I liked the old division on two lines better. Here, it should be 3 lines.

File: source/draw/gpu/opengl/OpenGLBgra.ooc (hunk 2 above)
Comment (jonathanudd) [inline source/draw/gpu/opengl/OpenGLBgra.ooc]: It will not be removed but instead import backend/GLTexture or something similar.

**Comment by davidhesselbom** [comment]
There are still conflicts. Also, don't merge with master yourself.

File: test/draw/gpu/pc/GpuSurfaceTest.ooc (hunk 3 above)
Comment (jonathanudd) [inline test/draw/gpu/pc/GpuSurfaceTest.ooc]: It's still valid.

**Comment by jonathanudd** [comment]
I have now rebased instead and solved the conflicts. The travis test fails on SynchronizedQueue but it works when I run it on my computer, why is that?

**Comment by davidhesselbom** [comment]
I don't know why, but it happens for me locally as well, about 5/100 times. On Travis is happens more often. There's an issue for this, but not here, so I'm going to close that one and open a new one here.

File: test/draw/gpu/pc/GpuSurfaceTest.ooc (hunk 3 above)
Comment (thomasfanell) [inline test/draw/gpu/pc/GpuSurfaceTest.ooc]: I think it should be safe the change this setting to 1.0f permanently, I thought @emilwestergren had already done so in his GPU cleanup branch?

Comment (emilwestergren) [inline test/draw/gpu/pc/GpuSurfaceTest.ooc]: The plan is to be able to set the point size in the GPU API, this is not possible today. The point size is set to 1.0f however. I think the reason for this comment is to be clear that in this test a point size of 1.0f is expected, so when we can change it in the API we should set it to 1.0f in this test.

Comment (thomasfanell) [inline test/draw/gpu/pc/GpuSurfaceTest.ooc]: Yes, that is correct.

**Comment by davidhesselbom** [comment]
It could be due to a race condition, maybe.

**Pull request merged**